src/modules/anon_chat/index.js

```diff
@@ -0,0 +1,66 @@
+const settings = require('../../settings');
+const watcher = require('../../watcher');
+const twitch = require('../../utils/twitch');
+
+let ignoreNextDC = false;
+const forcedURL = window.location.search.includes('bttv_anon_chat=true');
+
+class AnonChatModule {
+    constructor() {
+        settings.add({
+            id: 'anonChat',
+            name: 'Anon Chat',
+            defaultValue: false,
+            description: 'Join channels without appearing in chat'
+        });
+
+        watcher.on('load.chat', () => this.load());
+        watcher.on('chat.message', ($el, emberView) => this.onMessage($el, emberView));
+        settings.on('changed.anonChat', () => this.load());
+    }
+
+    part() {
+        const currentUser = twitch.getCurrentUser();
```

Comment:
pull all this logic out to a helper function that takes a nickname.. have this function send the message and call that